File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataDecimal.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Decimal import Decimal
from decimal import Decimal as D

@PythonTaskInfo
class DataDecimal(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in", data_type=Decimal)
        self.out = OutputPort(name="out", data_type=Decimal)

    def execute(self, grp):
        in1Data = self.in_.get_data(Decimal)
        dec = in1Data.values[0] * D(10)
        self.out.put_data(Decimal(dec))

File: com.systemincloud.examples.tasks.rtask/src/test/r/data/DataDecimal.R

library(R6)
library(sicrtask)

library(Rmpfr)

DataDecimal <- R6Class("DataDecimal",
  inherit = RTask,
  public = list(
    rtaskinfo = RTaskInfo$new(),
    in_ = NA,
    out = NA,
    initialize_ports = function() {
      self$in_ <- InputPort$new(name = "in", data_type = "Decimal")
      self$out <- OutputPort$new(name = "out", data_type = "Decimal")
    },
    initialize = function() {
      self$initialize_ports()
    },
    execute = function(grp) {
	  inData <- self$in_$get_data("Decimal")
	  dec <- in1Data$values[0] * mpfr(10)
	  self$out$put_data(Decimal$new(dec))
    }
  ),
  private = list(
  )
)
